### PR TITLE
configure Desktop logging

### PR DESCRIPTION
**Pull request by DhariniJeeva**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/JIRA_TICKET_NUMBER)

## Description

This PR installs `electron log` to add logs to desktop client.  

## How to Test

- run binary locally and perform search on DC 
- you can see the log file being generated in the below mentioned location based on your OS 

on Linux: ~/.config/{app name}/logs/main.log
on macOS: ~/Library/Logs/{app name}/main.log
on Windows: %USERPROFILE%\AppData\Roaming\{app name}\logs\main.log

:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:

- [ ] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: addons/api/package.json [hunk 1]
```diff
@@ -30,7 +30,6 @@
     "ember-cli-mirage": "^2.4.0",
     "ember-data": "~4.12.0",
     "ember-fetch": "^8.1.1",
-    "ember-auto-import": "^2.6.3",
```
Comment (calcaide) [inline addons/api/package.json:33]: Are we deleting `ember-auto-import`?

File: ui/desktop/electron-app/src/services/client-daemon-manager.js [hunk 2]
```diff
@@ -154,6 +156,8 @@ const searchCliCommand = (requestData) => {
   }
   const sanitizedToken = sanitizer.base62EscapeAndValidate(requestData.token);
 
+  log.info('logging search command:', searchCommand);
```
Comment (calcaide) [inline ui/desktop/electron-app/src/services/client-daemon-manager.js:159]: I think, before we log `searchCommand` we should delete the `token` field from it, we can not log sensitive data such as tokens.

**Comment by calcaide** [comment]
I will suggest to add a log when we detect the clientDaemon is failing and we fall back to the controller, [not sure should be here](https://github.com/hashicorp/boundary-ui/blob/main/addons/api/addon/handlers/client-daemon-handler.js#L119-L139), but somewhere around there.
The idea of my suggestion is to know:

- When and how often the daemon fails and we need to fallback to the controller.
- To know if we are fetching from daemon or controller.

File: addons/api/package.json (hunk 1 above)
Comment (calcaide) [reply addons/api/package.json:33]: Good catch!

File: ui/desktop/electron-app/src/services/client-daemon-manager.js (hunk 2 above)
Comment (calcaide) [reply ui/desktop/electron-app/src/services/client-daemon-manager.js:159]: Taking as example a line of the generated log:
```
{
  method: 'GET',
  path: 'http://internal.boundary.local/v1/search?force_refresh=true&query=&auth_token_id=at_JhnjID5a06&resource=targets',
  socketPath: '/Users/ccorvo/.boundary/socket/daemon.sock'
}
```

Inside the path, I see the `auth_token_id` is this something safe to log?

**Comment by calcaide** [comment]
Awesome!! this is looking great @DhariniJeeva !!

**Comment by ZedLi** [comment]
Is this going to get merged into main? What was the plan for this?

**Comment by calcaide** [comment]
Ups!! good catch @ZedLi , definetly this should be in a feature or LL branch, so we can keep working on it, build experimental binaries if need until we decide to bring it to `main`

**Comment by DhariniJeeva** [comment]
I wasn't sure when we would merge this or what the scope of this work is, but for now, as Carlos suggested, I will create an LLB for this work.

**Comment by ZedLi** [comment]
Closed and handled in https://github.com/hashicorp/boundary-ui/pull/2234

**Pull request closed without merging**